ROADMAP.md

```diff
@@ -0,0 +1,123 @@
+
+With that in mind, below we detail some of our short, mid and long-term goals. Most of these will be worked on concurrently, and reaching them will be a gradual process. The timeframes are also not set in stone, but serve as a tentative target the team is aiming for.
+
+
+Short-term goals
+----------------
+
+These are goals achievable within 3-6 months, tentatively done by Q3 2022.
+
+- **Stabilize the existing functionality and fix any major, high-impact and blocking bugs.**<br>
+  Tool stability is something we take very seriously and users shouldn't run into any major showstopping issues.
+
+  *How will we achieve this?*<br>
+  By manually testing the application in different scenarios, adding more automated tests, and addressing reports from internal and external users.
+  Any issues found that have a considerable user impact will be given highest priority and fixed ASAP.
+
+  *Definition of Done*<br>
+  This is difficult to gauge, but at some point we should stop running into any major issues on a frequent basis and should consider this goal as reached.
+
+
+- **Limited Alpha deployment and testing of the extension in k6 Cloud.**<br>
+
+  *How will we achieve this?*<br>
+  Soon we will create a small scale deployment for internal use and specific customers, meant to test the basic integration of the extension in k6 Cloud.
+
+  *Definition of Done*<br>
+  When the extension is usable in k6 Cloud; i.e. scripts can run and test results are shown. Frontend changes are not required at this stage.
+
+
+- **Update [API documentation](https://k6.io/docs/javascript-api/xk6-browser/).**<br>
+  The current documentation is not up-to-date which makes using xk6-browser outside of the examples quite difficult.
+
+  *How will we achieve this?*<br>
+  We are prioritizing working on documentation in all upcoming cycles.
+
+  *Definition of Done*<br>
+  When the documentation reflects the state of the current API.
+
+
+Mid-term goals
+--------------
+
+These are goals achievable within 6-12 months, tentatively done by mid 2023.
+
+- **Transition our API to be async/`Promise` based.**<br>
+  Currently (April 2022) most of our API is synchronous. This is due to the historical fact that k6 didn't support async behavior because of a missing per VU event loop, which is [now available](https://github.com/grafana/k6/pull/2228).
+  Async APIs are important for a browser testing tool, since most browser behavior and [CDP](https://chromedevtools.github.io/devtools-protocol/) (the protocol we use to communicate with the browser) is event-based. In order to implement this missing functionality and reach feature parity with tools like Playwright or Puppeteer, we need to expose an async API.
+
+  *How will we achieve this?*<br>
+  By gradually transitioning the current API to an async implementation, and implementing new features as async.
+
+  *Definition of Done*<br>
+  When most of the API can be used asynchronously.
```

Comment:
```suggestion
These are goals achievable within 6-12 months, tentatively done by mid-2023.

- **Transition our API to be async/`Promise` based.**<br>
  Currently (April 2022), most of our API is synchronous. This is due to the historical fact that k6 didn't support async behavior because of a missing per VU event loop, which is [now available](https://github.com/grafana/k6/pull/2228).
  Async APIs are important for a browser testing tool since most browser behavior and [CDP](https://chromedevtools.github.io/devtools-protocol/) (the protocol we use to communicate with the browser) is event-based. We need to expose an async API to implement this missing functionality and reach feature parity with tools like Playwright or Puppeteer.

  *How will we achieve this?*<br>
  By gradually transitioning the current API to an async implementation and implementing new features as async.

  *Definition of Done*<br>
  When most of the API can be used asynchronously.
```